mne/io/base.py

```diff
@@ -2422,6 +2428,28 @@ def concatenate_raws(raws, preload=None, events_list=None, verbose=None):
     events : ndarray of int, shape (n_events, 3)
         The events. Only returned if ``event_list`` is not None.
     """
+    _check_option(parameter='on_mismatch', value=on_mismatch,
+                  allowed_values=('raise', 'warn', 'ignore'))
+
+    first_dev_head_t = raws[0].info['dev_head_t']
+    for raw in raws[1:]:
```

Comment:
Can you factorize the code with concatenate_epochs ?

Reply:
@agramfort I looked into this but we're going so much epochs-specific magic in `concatenate_epochs()` that I felt this would actually complicate things

Reply:
No, wait. I think I can (re)use `_compare_epochs_infos()`. Need to look into this.

Reply:
I expect there to be some pain involved (maybe a new param for `concatenate_epochs`, maybe another BUG entry, etc.), but agree with @agramfort that at least the info-checking functionality should be shared between the two. I don't in principle see why the info comparisons should differ. Let's see if you agree from looking at the code, I don't know how they are different...

Reply:
I've tried to address your comments in b087b5b